### PR TITLE
feat: implement data ingest

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Notes
This PR integrates the data ingest functionality and is aimed at enhancing the functionality of our current geospatial data processing capabilities. By leveraging STAC, we can standardize the way geospatial data is indexed and accessed, improving interoperability and scalability of our data resources. It allows for the deployment of new assets including the data catalog ingest, API, and intake components through our CDK constructs.

### Checklist

Before you submit a pull request, please make sure you have the following:
- [x] Code changes are compact and well-structured to facilitate easy review
- [x] Changes are documented in the README.md and other relevant documentation pages
- [x] PR title and description accurately reflect the changes and are detailed enough for historical tracking
- [x] PR contains tests that cover all new code and the code has been manual tested
- [x] All new dependencies are declared (if any), and no unnecessary libraries are added
- [x] Performance impacts (if any) of the changes are evaluated and documented
- [x] Security implications of the changes (if any) are reviewed and addressed
- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md) and agree to follow the [Code of Conduct](../CODE_OF_CONDUCT.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/osml/data_catalog/dc_ingest_container.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+/**
+ * Represents the configuration for the DCIngestContainer Construct.
+ */
+export class DCIngestContainerConfig {
+  /**
+   * Creates an instance of DCIngestContainerConfig.
+   * @param {string} [DC_INGEST_CONTAINER="awsosml/awsosml/osml-data-intake-ingest:latest"] - Container image pull.
+   * @param {string} [DC_INGEST_BUILD_PATH="lib/osml-data-intake"] - The build path for the source code.
+   * @param {string} [DC_INGEST_BUILD_TARGET="osml_data_intake_ingest"] - The build target within the Dockerfile.
```
Comment (RanbirAulakh) [inline lib/osml/data_catalog/dc_ingest_container.ts:21]: Update the build target name

File: lib/osml/data_catalog/dc_ingest_container.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+ * Interface representing the properties for the DCIngestContainer construct.
+ */
+export interface DCIngestContainerProps {
+  /**
+   * The OSML deployment account.
+   */
+  account: OSMLAccount;
+
+  /**
+   * Optional flag to instruct building model runner container from source.
```
Comment (RanbirAulakh) [inline lib/osml/data_catalog/dc_ingest_container.ts:44]: Update `model runner`

File: lib/osml/data_catalog/dc_ingest_container.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+   */
+  buildFromSource?: boolean;
+
+  /**
+   * The OSML vpc to deploy into
+   */
+  osmlVpc: OSMLVpc;
+
+  /**
+   * Optional configuration for the DCContainer Construct
```
Comment (RanbirAulakh) [inline lib/osml/data_catalog/dc_ingest_container.ts:54]: Update `DCContainer` to `DCIngestContainer`

File: lib/osml/data_catalog/dc_ingest_container.ts (hunk 1 above)
Comment (RanbirAulakh) [reply lib/osml/data_catalog/dc_ingest_container.ts:21]: Also update the ingest repository below

File: lib/osml/data_catalog/dc_stac_container.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+/**
+ * Represents the configuration for the DCStacContainer Construct.
+ */
+export class DCStacContainerConfig {
+  /**
+   * Creates an instance of DCStacContainerConfig.
+   * @param {string} [DC_STAC_CONTAINER="awsosml/awsosml/osml-data-intake-stac:latest"] - Container image pull.
+   * @param {string} [DC_STAC_BUILD_PATH="lib/osml-data-intake"] - The build path for the source code.
+   * @param {string} [DC_STAC_BUILD_TARGET="osml_data_intake_stac"] - The build target within the Dockerfile.
```
Comment (RanbirAulakh) [inline lib/osml/data_catalog/dc_stac_container.ts:21]: Update build target and repository

File: lib/osml/data_catalog/dc_stac_container.ts [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    public DC_STAC_BUILD_PATH: string = "lib/osml-data-intake/",
+    public DC_STAC_CONTAINER: string = "awsosml/osml-data-intake-stac:latest",
+    public DC_STAC_BUILD_TARGET: string = "stac",
+    public DC_STAC_REPOSITORY: string = "data-intake-stac-container",
+    public DC_STAC_DOCKERFILE: string = "docker/Dockerfile.stac"
+  ) {}
+}
+
+/**
+ * Interface representing the properties for the DCFastAPIContainer construct.
```
Comment (RanbirAulakh) [inline lib/osml/data_catalog/dc_stac_container.ts:35]: Is it DCFastAPIContainer or DCStacContainer

File: lib/osml/data_catalog/dc_stac_container.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+ */
+export interface DCStacContainerProps {
+  /**
+   * The OSML deployment account.
+   */
+  account: OSMLAccount;
+
+  /**
+   * Optional flag to instruct building model runner container from source.
```
Comment (RanbirAulakh) [inline lib/osml/data_catalog/dc_stac_container.ts:44]: update `model runner` to data catalog stac

File: lib/osml/data_catalog/dc_stac_container.ts [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+   */
+  buildFromSource?: boolean;
+
+  /**
+   * The OSML vpc to deploy into
+   */
+  osmlVpc: OSMLVpc;
+
+  /**
+   * Optional configuration for the DCContainer Construct
```
Comment (RanbirAulakh) [inline lib/osml/data_catalog/dc_stac_container.ts:54]: Update construct name

File: lib/osml/data_intake/di_container.ts [hunk 8]
```diff
@@ -19,12 +19,14 @@ export class DIContainerConfig {
    * @param {string} [DI_CONTAINER="awsosml/osml-data-intake:latest"] - Container image to use for the lambda container.
    * @param {string} [DI_BUILD_PATH="lib/osml-data-intake"] - The build path for the Data Intake source code.
    * @param {string} [DI_BUILD_TARGET="osml_data_intake"] - The build target for the Data Intake Dockerfile.
+   * @param {string} [DI_DOCKERFILE="docker/Dockerfile.intake"] - The Dockerfile to build the container.
```
Comment (RanbirAulakh) [inline lib/osml/data_intake/di_container.ts:22]: Update build target

File: lib/osml/data_intake/di_dataplane.ts [hunk 9]
```diff
@@ -132,14 +134,14 @@ export class DIDataplane extends Construct {
 
     // Create a test output bucket
     this.outputBucket = new OSMLBucket(this, `DIInputBucket`, {
-      bucketName: props.account.id + "-" + this.config.S3_OUTPUT_BUCKET_NAME,
+      bucketName: this.config.S3_OUTPUT_BUCKET_NAME + "-" + props.account.id,
```
Comment (RanbirAulakh) [inline lib/osml/data_intake/di_dataplane.ts:137]: Just nit picking, could you update this line to

```
bucketName: `${this.config.S3_OUTPUT_BUCKET_NAME}-${props.account.id}`
```

File: lib/osml/osml_restapi.ts [hunk 10]
```diff
@@ -56,7 +56,6 @@ export interface OSMLRestApiProps {
 export class OSMLRestApi extends Construct {
   public requestAuthorizer: RequestAuthorizer;
   public restApi: RestApi;
-  public integration: Integration;
```
Comment (RanbirAulakh) [inline lib/osml/osml_restapi.ts:59]: Why is this removed? RestAPI needs an integration (for instance, Lambda function)

**Comment by RanbirAulakh** [comment]
Could you please update the commit message to reflect the title of this PR? Thanks!

File: lib/osml/data_catalog/dc_stac_container.ts (hunk 5 above)
Comment (drduhe) [reply lib/osml/data_catalog/dc_stac_container.ts:35]: StacContainer

File: lib/osml/osml_restapi.ts (hunk 10 above)
Comment (drduhe) [reply lib/osml/osml_restapi.ts:59]: It wasn't being used or set on the construct

Comment (RanbirAulakh) [reply lib/osml/osml_restapi.ts:59]: Oh, I was thinking it was coming from props.

File: lib/osml/data_intake/di_dataplane.ts [hunk 11]
```diff
@@ -25,21 +25,23 @@ export class DIDataplaneConfig {
...
     public LAMBDA_ROLE_NAME: string = "DILambdaRole",
     public LAMBDA_FUNCTION_NAME: string = "DILambda",
     public LAMBDA_MEMORY_SIZE: number = 4096,
+    public LAMBDA_RETRY_ATTEMPTS: number = 3,
     public LAMBDA_STORAGE_SIZE: number = 10,
     public LAMBDA_TIMEOUT: number = 120,
-    public SNS_INPUT_TOPIC_NAME: string = "DIInputTopic",
-    public SNS_OUTPUT_TOPIC_NAME: string = "DIOutputTopic",
+    public SNS_INPUT_TOPIC_NAME: string = "osml-data-intake",
+    public SNS_STAC_TOPIC: string = "osml-stac-ingest",
```
Comment (RanbirAulakh) [inline lib/osml/data_intake/di_dataplane.ts:44]: This somewhat breaks the naming pattern as other classes. I would prefer

- DIInputTopic or DIIntakeTopic
- DIStacTopic

Comment (drduhe) [reply lib/osml/data_intake/di_dataplane.ts:44]: yeah I realized those naming patterns are actually non conventional and I was going to spin up a backlog to go back and update them

Comment (RanbirAulakh) [reply lib/osml/data_intake/di_dataplane.ts:44]: One has `_NAME` and other doesn't. Could you pick one? Other than that, I would be a fan if we append `-topic` at the end of the string.

Comment (drduhe) [reply lib/osml/data_intake/di_dataplane.ts]: Updated to have _NAME - not adding -topic to reduce redundant characters.

**Review by RanbirAulakh — APPROVED** [review]
